File: flutter_new_app/lib/models/api/finance_history.dart
import '../enums/finance_type.dart';

class FinanceHistory {
  const FinanceHistory({required this.itemName, required this.itemValue, required this.categoryShopId, required this.accountId, required this.type});

  final String itemName;
  final FinanceHistoryType type;
  final num itemValue;
  final num categoryShopId;
  final num accountId;

  factory FinanceHistory.fromJson(Map<String, dynamic> json) {
    return FinanceHistory(
        itemName: json['itemName'] as String,
        type: FinanceHistoryType.values[json['type'] as int],
        itemValue: json['itemValue'] as num,
        categoryShopId: json['categoryShopId'] as num,
        accountId: json['accountId'] as num);
  }

  Map<String, dynamic> toJson() => {
    'itemName': itemName,
    'itemValue': itemValue,
    'type': type.index,
    'categoryShopId': categoryShopId,
    'accountId': accountId,
  };
}

File: SwiftHelperBackend/SwiftHelperBackend.DAL/Models/FinanceHistory.cs
using SwiftHelperBackend.DAL.Models.Enums;

namespace SwiftHelperBackend.DAL.Models;

public class FinanceHistory : IdBase
{
    public string? ItemName { get; set; } = null!;
    public decimal ItemValue { get; set; }
    public DateTime CreatedDate { get; set; }
    public FinanceHistoryType Type { get; set; } 
    
    public long CategoryShopId { get; set; }
    public long AccountId { get; set; }
    public string UserId { get; set; } = null!;
    
    public ApplicationUser User { get; set; } = null!;
    public FinanceAccount Account { get; set; } = null!;
    public FinanceShopCategory Category { get; set; } = null!;
}